app/Jobs/Warnings/WarningFetcher.php

```diff
@@ -70,62 +99,46 @@ private function fetchIpmaWarnings(): bool
-                $endedAt = $this->carbonise($data['endTime'])->toDateTimeString();
-
-                $warningExists = IpmaWarning::query()
-                    ->where('area_id', $data['idAreaAviso'])
-                    ->where('awareness_level_id', $awarenessLevelId)
-                    ->where('is_active', 1)
-                    ->where('started_at', '<=', $startedAt)
-                    ->where('ended_at', '>', $startedAt)
-                    ->where('started_at', '<', $endedAt)
-                    ->where('ended_at', '>=', $endedAt)
-                    ->first();
-
-                if (! $warningExists) {
-                    $areasMap = config('ipma.areas', []);
-
-                    $warning = new IpmaWarning();
-                    $warning->uuid = Uuid::uuid4();
-                    $warning->text = ! empty($data['text']) ? $data['text'] : null;
-                    $warning->awareness_type_name = $data['awarenessTypeName'];
-                    $warning->zone_name = Arr::get($areasMap, $data['idAreaAviso'].'.zone');
-                    $warning->area_id = $data['idAreaAviso'];
-                    $warning->area_name = Arr::get($areasMap, $data['idAreaAviso'].'.area');
-                    $warning->awareness_level_id = $awarenessLevelId;
-                    $warning->started_at = $startedAt;
-                    $warning->ended_at = $endedAt;
-                    $warning->save();
-
-                    $this->logger->info(\sprintf(
-                        'Created Ipma warning %s',
-                        $warning->id
-                    ));
+        $ipmaWarnings = collect($response['data'])
+            ->filter(function ($value) {
+                if (! empty($value['awarenessLevelID'])) {
+                    $awarenessLevelId = \mb_strtolower($value['awarenessLevelID']);
+                    $awarenessIsAllowed = \in_array($awarenessLevelId, self::ALLOWED_AWARENESS_LEVELS, true);
+                    if ($awarenessIsAllowed) {
+                        return $value;
+                    }
                 }
+            })->map(function ($item) {
+                // get aux data
+                $countyId = Arr::get(self::AREA_CODES_MAP, $item['idAreaAviso']);
+                $county = County::find($countyId);
+                $district = $county->district;
+                $region = ! \in_array($district->id, [19, 20], true) ? 'CONTINENTE' : $district->name;
+
+                // set/modify attributtes
+                $item['id'] = Uuid::uuid4()->toString();
+                $item['region'] = $region;
+                $item['county'] = $county->name;
+                $item['started_at'] = $this->carbonise($item['startTime'])->toDateTimeString();
+                $item['ended_at'] = $this->carbonise($item['endTime'])->toDateTimeString();
+
+                return $item;
             });
+
+        if (Cache::has('ipma_warnings')) {
```

Comment:
Use the `remember()` method, instead.

Reply:
the remember method makes sense for the controller not for the fetcher, otherwise there is the risk of having no data in the cache if the cron and the cache lifetime get desyncronized. In my POV the fetcher should delete the cache and add it. The remember is more to retrieve actions like the controller.